backend/coins/btc/transaction_test.go

```diff
@@ -125,3 +142,82 @@ func TestGetFeePerKb(t *testing.T) {
 	}
 
 }
+
+func TestPickChangeAddressSucceeds(t *testing.T) {
+	rootFingerprint := []byte{1, 2, 3, 4}
+	signingConfigurations := signing.Configurations{
+		signing.NewBitcoinConfiguration(signing.ScriptTypeP2WPKHP2SH, rootFingerprint, mustKeypath(t, "m/49'/0'/1'"), test.TstMustXKey("xpub6CUmEcJb7juvpvNs2hKMc9BP1n82ixzUb4jyHUdYzSLmnXru3nb4hhGsfS23WRx8hgJLxMxZ7WcBGzTiYfiANUQZe3TVFghLrxvA2Ls7u4a")),
+	}
+
+	testCases := []struct {
+		name                      string
+		signingConfigurationsFunc func() signing.Configurations
+		utxos                     map[wire.OutPoint]maketx.UTXO
+		wantAddress               string
+	}{
+		{
+			name:        "One subaccount",
+			wantAddress: "2MsjfL7qJTBDnuhUiDJEUR9iWhtcyEnzQYU",
+		},
+		{
+			name: "p2wpkh",
+			signingConfigurationsFunc: func() signing.Configurations {
+				return append(signingConfigurations,
+					signing.NewBitcoinConfiguration(signing.ScriptTypeP2WPKH, rootFingerprint, mustKeypath(t, "m/84'/0'/1'"), test.TstMustXKey("xpub6Cxa67Bfe1Aw7YVtdqKPYLhSkf7omb7WkGXQzof15VXbAZKVct1caHHK55UQN2Fnojbp2okiBCbGXyQSRzMQ6XKJJeeM2jAt6FR8K8ckA88")),
+				)
+			},
+			wantAddress: "tb1q42x65gqluatm6x6vvwlpg93wqcvz8jn3arakke",
+		},
+		{
+			name: "multiple accounts, no p2wpkh or p2tr",
+			signingConfigurationsFunc: func() signing.Configurations {
+				return append(signing.Configurations{
+					signing.NewBitcoinConfiguration(signing.ScriptTypeP2WPKH, rootFingerprint, mustKeypath(t, "m/84'/2'/1'"), test.TstMustXKey("xpub6DReBHtKxgeZJJrrhPEHz9kzEZU1BaQ4kPQ2J1tfjA9DMBKT2bor1ynoAPCsxdyJyZrYK5YsYmkknV5KPtpKeVb2HMX6iQ9wjpAhNSANGiA")),
```

Comment:
the name says "no pwpkh", but this is a P2WPKH subaccount.

Reply:
Yeah good catch, fixed!